File: Metode-Avansate-de-Programare/lab2/factory/TaskContainerFactory.java
package sem1_2.factory;

import sem1_2.model.Container;
import sem1_2.model.StackContainer;
import sem1_2.model.QueueContainer;

public class TaskContainerFactory implements Factory {
    private static TaskContainerFactory instance = new TaskContainerFactory();
    private TaskContainerFactory() {}
    public static TaskContainerFactory getInstance() {
        return instance;
    }
    @Override
    public Container createContainer(Strategy strategy) {
        if (strategy == Strategy.FIFO) {
            return new QueueContainer();
        } else {
            return new StackContainer();
        }
    }
}

File: Metode-Avansate-de-Programare/lab11/lab11/TaskContainerFactory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab11;

public class TaskContainerFactory : Factory
{
    private static TaskContainerFactory container = null;

    private TaskContainerFactory()
    {
    }

    public static TaskContainerFactory getInstance()
    {
        if (container == null)
        {
            container = new TaskContainerFactory();
        }

        return container;
    }

    public override Container createContainer(Strategy strategy)
    {
        if (strategy == Strategy.FIFO)
        {
            return new QueueContainer();
        }
        else if (strategy == Strategy.LIFO)
        {
            return new StackContainer();
        }

        return null;
    }
}